spec/system/admin/products_v3/products_spec.rb

```diff
@@ -26,20 +26,41 @@
   describe "sorting" do
     let!(:product_b) { create(:simple_product, name: "Bananas") }
     let!(:product_a) { create(:simple_product, name: "Apples") }
+    let(:products_table) { "table.products" }
 
     before do
       visit admin_products_url
     end
 
-    it "Should sort products alphabetically by default" do
-      within "table.products" do
+    it "Should sort products alphabetically by default in ascending order" do
+      within products_table do
         # Gather input values, because page.content doesn't include them.
         input_content = page.find_all('input[type=text]').map(&:value).join
 
         # Products are in correct order.
         expect(input_content).to match /Apples.*Bananas/
       end
     end
+
+    context "when clicked on 'Name' column header" do
+      it "Should sort products alphabetically in descending/ascending order" do
+        within products_table do
+          name_header = page.find('th > a[data-column="name"]')
+
+          # Sort in descending order
+          name_header.click
+          sleep(0.2)
```

Comment:
It requires some delay here otherwise the `page` doesn't have the updated content.

I think it's something to do with Turbo, as it doesn't technically load the entire page and changes are updated in the existing page. So there's no way for Capybara to know whether the contents are fully loaded or not and it just moves on without waiting enough. 😅

Reply:
Actually, I think it's because capybara methods will wait until the specified element exists. Eg `expect(page).to have_content /Bananas.*Apples/` would wait. 

But this code doesn't know how to wait. We need something like:
```
expect(page).to have_content(/Bananas.*Apples/, include_input_values: true)
```

But I'm not sure it's worth trying to create that. How about instead of sleep, we wait for:
```suggestion
          expect(page).to have_content("Name ▼") # this indicates the re-sorted content has loaded
```

Reply:
Thank you so much, David. It worked ❤️